src/buffer.rs

```diff
@@ -71,16 +71,31 @@ impl<'hw> Buffer<'hw> {
     /// # Returns
     ///
     /// A Reference to the wrapped `Hardware` object.
-    pub(crate) fn hardware(&self) -> &'hw RefCell<dyn Hardware> {
+    pub fn hardware(&self) -> &'hw RefCell<dyn Hardware> {
         self.hardware
     }
 
+    /// Returns the size of the allocated memory.
+    ///
+    /// # Returns
+    ///
+    /// The size of the allocated memory.
+    pub fn size(&self) -> usize {
+        self.size
+    }
+
     /// Returns the const handle owned by this buffer.
     ///
     /// # Returns
     ///
     /// Owned handle as a const pointer.
-    pub(crate) unsafe fn as_handle(&self) -> *const u8 {
+    ///
+    /// # Safety
+    ///
+    /// This function returns a raw pointer of the inner memory or a haandle of the associated
+    /// hardware-specific object.
+    /// The value can not be used without knowing the associated hardware.
```

Comment:
I think the returned value would be clearer.
```suggestion
    /// The returned value can not be used without knowing the associated hardware.
```